### PR TITLE
fix: make version setting script work with both main and flask

### LINKED ISSUE #9770: [Bug]: Create release action not working
### Describe the bug

Running the `SEMVER_VERSION=x.x.x VERSION_NUMBER=xxxx yarn create-release` script fails (replace x by digits)

```
Run SEMVER_VERSION=7.24.0 VERSION_NUMBER=1233 yarn create-release
  SEMVER_VERSION=7.[2](https://github.com/MetaMask/metamask-mobile/actions/runs/9252637241/job/25450637222#step:5:2)4.0 VERSION_NUMBER=1233 yarn create-release
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
yarn run v1.22.22
$ ./scripts/set-versions.sh && yarn update-changelog
./scripts/set-versions.sh: line 8[5](https://github.com/MetaMask/metamask-mobile/actions/runs/9252637241/job/25450637222#step:5:6): PRODUCTION_BUILD_VERSION_NUMBER: unbound variable
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

See for instance this run https://github.com/MetaMask/metamask-mobile/actions/runs/9252637241/job/25450637222

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

run `SEMVER_VERSION=x.x.x VERSION_NUMBER=xxxx yarn create-release`

### Error messages or log output

_No response_

### Version

any

### Build type

None

### Device

any

### Operating system

Other (please elaborate in the "Additional Context" section)

### Additional context

not OS specific

### Severity

_No response_

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

- Fix the release number search and replace for Flask
- fix the package.json parsing with Jq and sponge (thanks @legobeat)
- fix sed -e commands to be more robust
  - make the syntax specifically work for Ubuntu Sed version as it's the CI target (thanks @legobeat)
  - add capture group and back ref to prevent repeating the key string
  - check for spaces before the key (that's what actually broke the script as `VERSION_NUMBER` matches `VERSION_NUMBER` and `FLASK_VERSION_NUMBER`. With a space before, there's no confusion)
- move sed -i command as it was creating backups and mistakingly use "-e" as the suffix
- Extract file names in constants
- improve console output messages to actually match steps
- add check that `VERSION_NUMBER` and `FLASK_VERSION_NUMBER` are similar before starting. They should be.
- remove dead code.

## **Related issues**

Fixes #9770 

## **Manual testing steps**

1. run `SEMVER_VERSION=7.24.0 VERSION_NUMBER=1334 ./scripts/set-versions.sh`
2. console output should be:
  ```
  VERSION_NUMBER and SEMVER_VERSION are valid.
  -------------------
  Updating files:
  - package.json updated
  - android/app/build.gradle updated
  - bitrise.yml updated
  - ios/MetaMask.xcodeproj/project.pbxproj updated
  -------------------
  files updated with:
  semver version: 7.24.0
  version number: 1334
  ``` 
3. Check with `git diff`: files should be updated with the correct semver version and version number

### **Before**

image

#### Console output
image

#### Git status
image

#### Git diff (sample screenshot)
image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: scripts/set-versions.sh [hunk 1]
```diff
@@ -28,31 +29,51 @@ log_and_exit () {
...
+  sed -e 's/\([[:space:]]*VERSION_NUMBER: \)[0-9]\+/\1'"$VERSION_NUMBER"'/' -i '' $BITRISE_YML_FILE
+  sed -e 's/\([[:space:]]*FLASK_VERSION_NUMBER: \)[0-9]\+/\1'"$VERSION_NUMBER"'/' -i '' $BITRISE_YML_FILE
+  echo "- $BITRISE_YML_FILE updated"
+
 
   # update ios/MetaMask.xcodeproj/project.pbxproj
-  sed -i -e 's/MARKETING_VERSION = .*/MARKETING_VERSION = '"$SEMVER_VERSION;"'/' ios/MetaMask.xcodeproj/project.pbxproj
-  sed -i -e 's/CURRENT_PROJECT_VERSION = [0-9]\+/CURRENT_PROJECT_VERSION = '"$VERSION_NUMBER"'/' ios/MetaMask.xcodeproj/project.pbxproj
+  sed -e 's/\([[:space:]]*MARKETING_VERSION = \).*/\1'"$SEMVER_VERSION;"'/' -i '' $IOS_PROJECT_FILE
+  sed -e 's/\([[:space:]]*CURRENT_PROJECT_VERSION = \)[0-9]\+/\1'"$VERSION_NUMBER"'/' -i '' $IOS_PROJECT_FILE
```
Comment (legobeat) [inline scripts/set-versions.sh:56]:
```suggestion
  sed -e 's/\([[:space:]]*MARKETING_VERSION = \).*/\1'"$SEMVER_VERSION;"'/' -i '' $IOS_PROJECT_FILE
  sed -e 's/\([[:space:]]*CURRENT_PROJECT_VERSION = \)[0-9]\+/\1'"$VERSION_NUMBER"'/' -i '' $IOS_PROJECT_FILE
```

```suggestion
  sed -Ei 's/(\s*MARKETING_VERSION = ).*/\1'"$SEMVER_VERSION;"'/' $IOS_PROJECT_FILE
  sed -Ei 's/(\s*CURRENT_PROJECT_VERSION = )[0-9]\+/\1'"$VERSION_NUMBER"'/' $IOS_PROJECT_FILE
```

Do we actually want to do `-i ''`? Omitting the suffix _should_ make it automatic with GNU sed. Also a bit more readable with `-E` imo and closer syntax to what we're used to from JS. The `-e`is redundant with a single expression.

File: scripts/set-versions.sh [hunk 2]
```diff
@@ -28,31 +29,51 @@ log_and_exit () {
...
 perform_updates () {
-  echo -e "creating release\nsemver version: $SEMVER_VERSION\nversion number: $VERSION_NUMBER"
-
   # update package.json
-  tmp="package.json_temp"
-  jq --arg semverVersion "$SEMVER_VERSION" '.version = $semverVersion' package.json > "$tmp"
-  mv "$tmp" package.json
+  tmp="${PACKAGE_JSON_FILE}_temp"
+  jq --arg semverVersion "$SEMVER_VERSION" '.version = $semverVersion' $PACKAGE_JSON_FILE > "$tmp"
+  mv "$tmp" $PACKAGE_JSON_FILE
```
Comment (legobeat) [inline scripts/set-versions.sh:35]:
```suggestion
  jq --arg semverVersion "$SEMVER_VERSION" '.version = $semverVersion' $PACKAGE_JSON_FILE | sponge $PACKAGE_JSON_FILE
```

Do we have [`sponge`](https://linux.die.net/man/1/sponge) ([`moreutils`](https://launchpad.net/ubuntu/trusty/+package/moreutils)) available?

File: scripts/set-versions.sh [hunk 3]
```diff
@@ -28,31 +29,51 @@ log_and_exit () {
...
-  sed -i -e 's/CURRENT_PROJECT_VERSION = [0-9]\+/CURRENT_PROJECT_VERSION = '"$VERSION_NUMBER"'/' ios/MetaMask.xcodeproj/project.pbxproj
+  sed -e 's/\([[:space:]]*MARKETING_VERSION = \).*/\1'"$SEMVER_VERSION;"'/' -i '' $IOS_PROJECT_FILE
+  sed -e 's/\([[:space:]]*CURRENT_PROJECT_VERSION = \)[0-9]\+/\1'"$VERSION_NUMBER"'/' -i '' $IOS_PROJECT_FILE
+  echo "- $IOS_PROJECT_FILE updated"
+
+  echo -e "-------------------"
+  echo -e "files updated with:"
+  echo -e "semver version: $SEMVER_VERSION"
+  echo -e "version number: $VERSION_NUMBER"
+
```
Comment (legobeat) [inline scripts/set-versions.sh:63]:
```suggestion
  cat <<EOT
- $IOS_PROJECT_FILE updated
-------------------
files updated with:
semver version: $SEMVER_VERSION
version number: $VERSION_NUMBER
EOT

```

(could) Matter of style I guess, but for many lines a heredoc reads nicer imo.

**Review by Andepande — DISMISSED** [review]
LGTM

File: scripts/set-versions.sh (hunk 2 above)
Comment (NicolasMassart) [reply scripts/set-versions.sh:35]: I don't know... will check

File: scripts/set-versions.sh (hunk 3 above)
Comment (NicolasMassart) [reply scripts/set-versions.sh:63]: I prefer to leave it like this for now.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**